dashboard/client/src/pages/node/NodeRow.tsx

```diff
@@ -0,0 +1,272 @@
+
+  return (
+    <TableRow key={hostname + rowIndex}>
+      <TableCell>
+        <IconButton size="small" onClick={onExpandButtonClick}>
+          {!expanded ? (
+            <AddIcon className={classes.expandCollapseIcon} />
+          ) : (
+            <RemoveIcon className={classes.expandCollapseIcon} />
+          )}
+        </IconButton>
+      </TableCell>
+      <TableCell>
+        <StatusChip type="node" status={raylet.state} />
+      </TableCell>
+      <TableCell align="center">
+        <Tooltip title={raylet.nodeId} arrow interactive>
+          <Link to={`/node/${raylet.nodeId}`} className={classes.idCol}>
+            {raylet.nodeId}
+          </Link>
+        </Tooltip>
+      </TableCell>
+      <TableCell align="center">{hostname}</TableCell>
+      <TableCell align="center">{ip}</TableCell>
+      <TableCell>
+        <PercentageBar num={Number(cpu)} total={100}>
+          {cpu}%
+        </PercentageBar>
+      </TableCell>
+      <TableCell>
+        <PercentageBar num={Number(mem[0] - mem[1])} total={mem[0]}>
+          {memoryConverter(mem[0] - mem[1])}/{memoryConverter(mem[0])}(
+          {mem[2].toFixed(1)}
+          %)
+        </PercentageBar>
+      </TableCell>
+      <TableCell>
+        {raylet && (
+          <PercentageBar
+            num={raylet.objectStoreUsedMemory}
+            total={objectStoreTotalMemory}
+          >
+            {memoryConverter(raylet.objectStoreUsedMemory)}/
+            {memoryConverter(objectStoreTotalMemory)}
+          </PercentageBar>
+        )}
+      </TableCell>
+      <TableCell>
+        {disk && disk["/"] && (
+          <PercentageBar num={Number(disk["/"].used)} total={disk["/"].total}>
+            {memoryConverter(disk["/"].used)}/{memoryConverter(disk["/"].total)}
+            ({disk["/"].percent.toFixed(1)}%)
+          </PercentageBar>
+        )}
+      </TableCell>
+      <TableCell align="center">{memoryConverter(networkSpeed[0])}/s</TableCell>
+      <TableCell align="center">{memoryConverter(networkSpeed[1])}/s</TableCell>
+      <TableCell>
+        <Link to={`/log/${encodeURIComponent(logUrl)}`}>Log</Link>
+      </TableCell>
+    </TableRow>
+  );
+};
+
+type WorkerRowProps = {
+  /**
+   * Details of the worker
+   */
+  worker: Worker;
+  /**
+   * Detail of the node the worker is inside.
+   */
+  node: NodeDetail;
+};
+
+/**
+ * A single row that represents the data of a Worker
+ */
+const WorkerRow = ({ node, worker }: WorkerRowProps) => {
+  const classes = useNodeRowStyles();
+
+  const { mem, logUrl } = node;
+  const {
+    pid,
+    cpuPercent: cpu = 0,
+    memoryInfo,
+    coreWorkerStats,
+    cmdline,
+  } = worker;
+
+  const coreWorker = coreWorkerStats.length ? coreWorkerStats[0] : undefined;
+  const workerLogUrl = coreWorker
+    ? `/log/${encodeURIComponent(logUrl)}?fileName=${coreWorker.workerId}`
+    : `/log/${encodeURIComponent(logUrl)}`;
+
+  return (
+    <TableRow key={pid}>
+      <TableCell></TableCell>
```

Comment:
maybe this empty TableCell can be removed?

Reply:
I understand what you mean. generally i return a component as a whole, such as return a <div> in a component, and when use the component in a 'map' return just add a 'kay' param. in this case it return a table cell which in a 'Fragment', I don't know whether it's a good way, but it's maked sense in here. Maybe it do not need to fix.

moreover, we export 'NodeRows' in this file, so the file named 'NodeRows' too?